compiler/src/dotty/tools/dotc/transform/ExplicitOuter.scala

```diff
@@ -276,7 +276,7 @@ object ExplicitOuter {
           if (tpe.prefix eq NoPrefix) cls.owner.enclosingClass.thisType
           else tpe.prefix
         case _ =>
-          outerPrefix(tpe.underlying)
+          outerPrefix(tpe.underlying(ctx.withPhaseNoLater(ctx.erasurePhase)))
```

Comment:
Since this is somewhat subtle, I would add a comment here that restates what is in the commit message.